present/markdown.py

```diff
@@ -497,9 +501,8 @@ def parse(self, text):
                         and obj["children"][0]["type"] == "image"
                     ):
                         image = obj["children"][0]
-                        image["src"] = os.path.join(
-                            self.file_dir, os.path.expanduser(image["src"])
-                        )
+                        image["src"] = os.path.join(self.dirname, image["src"])
```

Comment:
I think the `os.path.expanduser` is needed here for the `image["src"]`.

Assuming the following:
 * the markdown file is located in current working directory,
 * the image is located under: `/home/username/image.png`,
 * the image is added with `![](~/image.png)`.

The `os.path.join` would try to join current working directory (e.g. `/home/username/markdown_dir`) and `~/image.png` which would result in something like `/home/username/markdown_dir/~/image.png` and an error.

However, if we used `os.path.expanduser` it would change `~` to `/home/username` and then the path would be joined correctly (or left as the last element of `os.path.join` - `/home/username/image.png` in this case).

Reply:
I guess we can add `os.path.expanduser` if that supports your use case since the `os.path.join` will resolve correctly in both cases:

- Relative image path
  ```
  >>> os.path.join("/home/user/talk", os.path.expanduser("assets/image.png"))
  /home/user/talk/assets/image.png
  ```
- Absolute image path
  ```
  >>> os.path.join("/home/user/talk", os.path.expanduser("~/assets/image.png"))
  /home/user/assets/image.png
  ```

Reply:
I've pushed the change to master. Will make a release today.